src/components/select/select.js

```diff
@@ -1176,8 +1194,14 @@ function SelectProvider($$interimElementProvider) {
 
         function onMenuKeyDown(ev) {
           var keyCodes = $mdConstant.KEY_CODE;
-          ev.preventDefault();
-          ev.stopPropagation();
+          // If we have an input the md-select-header we want to allow
+          // users to type in it.
+          if (opts.selectCtrl.mdSelectHeaderInput === ev.target) {
```

Comment:
I'm unsure you want to leave this early. For example, if I were typing in a searchbox, I'd still want to be able to use the arrow keys to select items. This is a little tricky with maintaining focus, so it's fine if that is a future improvement or comes as part of your search component.